fusion/fusion/fieldextractor.h

```diff
@@ -0,0 +1,302 @@
+// HEADER PLACEHOLDER
+// contact Jeff Nye, jeffnye-gh
+//
+//! \file fieldextractor.h  mavis shim, extract fields from encodings
+#pragma once
+#include "fusionexceptions.h"
+#include "uArchInfo.h"
+#include "Inst.h"
+#include "Mavis.h"
+#include <vector>
+#include <optional>
+#include <ostream>
+#include <iostream>
+using namespace std;
+
+//! \class FieldExtractor
+//!
+//! \brief example class for extracting field values from mavis encodings
+//!
+//! This implementation provides a shim between operations typically
+//! used during fusion contraint checking and transformation and mavis.
+//!
+//! The intent is any alternative to FieldExtractor could be created
+//! if compliant with the interface. This has not been tested.
+//!
+//! It would also be interesting to determine if more of this could be
+//! delegated to existing methods in mavis.
+class FieldExtractor
+{
+  public:
+    //! \brief Part of mechanism to extract named fields from encodings
+    //!
+    //! OperandFieldID has no entry for immediates
+    //! \note I am overloading RS_MAX when referencing immediate fields
+    using FieldName = mavis::InstMetaData::OperandFieldID;
+    //! \brief alias used to extract Special fields by enum
+    using SFieldName = mavis::ExtractorIF::SpecialField;
+
+    //! \brief ...
+    using InstPtrType = Instruction<uArchInfo>::PtrType;
+    //! \brief ...
+    using InstPtrListType = std::vector<Instruction<uArchInfo>::PtrType>;
+    //! \brief type for getXRegs operations from mavis
+    using MavisBitMaskType = mavis::DecodedInstructionInfo::BitMask;
+    //! \brief type for getXRegs operations from mavis
+    using RegsGetter =
+        MavisBitMaskType (Instruction<uArchInfo>::*)() const;
+    //! \brief duplicated arguments do not need to be explicitly expressed
+    using OptArg = std::optional<FieldName>;
+
+    //!  \brief compairson function primitives
+    enum FUNC
+    {
+        EQ,
+        LT
+    };
+
+    //!  \brief placeholder
+    void info(std::ostream & os = std::cout)
+    {
+        os << "FieldExtractor info()" << std::endl;
+    }
+
+    //!  \brief extract value of named encoding field
+    //!
+    //! handles field name and immediate checking
+    //! \note RS_MAX is overloaded to identify get's for immediate fields
+    uint32_t getField(InstPtrType inst, FieldName f) const
```

Comment:
So, outside of complaining about missing `const`s ( 😆 ), I have a huge issue with single letter variable names for one main reason: search for `f` in this function.  😱

Reply:
fixed.